neqo-crypto/src/p11.rs

```diff
@@ -287,6 +287,46 @@ impl Item {
     }
 }
 
+unsafe fn destroy_secitem_array(array: *mut SECItemArray) {
+    SECITEM_FreeArray(array, PRBool::from(true));
+}
+scoped_ptr!(ItemArray, SECItemArray, destroy_secitem_array);
+
+impl<'a> IntoIterator for &'a ItemArray {
+    type Item = &'a [u8];
+    type IntoIter = ItemArrayIterator<'a>;
+    fn into_iter(self) -> Self::IntoIter {
+        if self.ptr.is_null() || unsafe { *self.ptr }.len == 0 {
+            Self::IntoIter {
+                iter: Iter::default(),
+            }
+        } else {
+            // (*self.ptr).items points to (*self.ptr).len > 0 consecutive elements of type SECItem
+            let items: &[SECItem] = unsafe {
+                #[allow(clippy::disallowed_methods)]
```

Comment:
It seems fairly trivial to change `null_safe_slice` to make it usable here...

```rust
unsafe fn null_safe_slice<'a, T, L>(data: *const T, len: L) -> &'a [T]
where
    usize: TryFrom<L>,
{
    if data.is_null() {
        &[]
    } else if let Ok(len) = usize::try_from(len) {
        if len == 0 {
            &[]
        } else {
            #[allow(clippy::disallowed_methods)]
            std::slice::from_raw_parts(data, len)
        }
    } else {
        panic!("null_safe_slice: size overflow");
    }
}
```

Reply:
done!

Reply:
Oh, I missed the `len == 0` check. I'll add it.